src/main/java/mekanism/tools/common/MekanismTools.java

```diff
@@ -111,172 +77,64 @@ public void init(FMLInitializationEvent event) {
+    private static void addSmeltingRecipe(List<ToolsItem> itemSet, ItemStack nugget) {
+        itemSet.forEach(toolsItem -> addSmeltingRecipe(toolsItem, nugget));
+    }
 
-        GameRegistry.addSmelting(ToolsItems.SteelPaxel, new ItemStack(MekanismItems.Nugget, 1, 4), 0.1F);
-        GameRegistry.addSmelting(ToolsItems.SteelPickaxe, new ItemStack(MekanismItems.Nugget, 1, 4), 0.1F);
-        GameRegistry.addSmelting(ToolsItems.SteelAxe, new ItemStack(MekanismItems.Nugget, 1, 4), 0.1F);
-        GameRegistry.addSmelting(ToolsItems.SteelShovel, new ItemStack(MekanismItems.Nugget, 1, 4), 0.1F);
-        GameRegistry.addSmelting(ToolsItems.SteelHoe, new ItemStack(MekanismItems.Nugget, 1, 4), 0.1F);
-        GameRegistry.addSmelting(ToolsItems.SteelSword, new ItemStack(MekanismItems.Nugget, 1, 4), 0.1F);
-        GameRegistry.addSmelting(ToolsItems.SteelHelmet, new ItemStack(MekanismItems.Nugget, 1, 4), 0.1F);
-        GameRegistry.addSmelting(ToolsItems.SteelChestplate, new ItemStack(MekanismItems.Nugget, 1, 4), 0.1F);
-        GameRegistry.addSmelting(ToolsItems.SteelLeggings, new ItemStack(MekanismItems.Nugget, 1, 4), 0.1F);
-        GameRegistry.addSmelting(ToolsItems.SteelBoots, new ItemStack(MekanismItems.Nugget, 1, 4), 0.1F);
+    private static void addSmeltingRecipe(ToolsItem toolsItem, ItemStack nugget) {
+        GameRegistry.addSmelting(toolsItem.getItemStackAnyDamage(), nugget, 0.1F);
     }
 
-    private void setStackIfEmpty(EntityLivingBase entity, EntityEquipmentSlot slot, ItemStack item) {
-        if (entity.getItemStackFromSlot(slot).isEmpty()) {
+    private void setStackIfCaseAndEmpty(EntityLivingBase entity, EntityEquipmentSlot slot, ItemStack item, boolean shouldSet) {
+        if (shouldSet && entity.getItemStackFromSlot(slot).isEmpty()) {
             entity.setItemStackToSlot(slot, item);
         }
     }
 
-    private void setEntityArmorWithChance(Random random, EntityLivingBase entity, Item sword, Item helmet, Item chestplate, Item leggings, Item boots) {
-        if (entity instanceof EntityZombie && random.nextInt(100) < 50) {
-            setStackIfEmpty(entity, EntityEquipmentSlot.MAINHAND, new ItemStack(sword));
-        }
-        if (random.nextInt(100) < 50) {
-            setStackIfEmpty(entity, EntityEquipmentSlot.HEAD, new ItemStack(helmet));
-        }
-        if (random.nextInt(100) < 50) {
-            setStackIfEmpty(entity, EntityEquipmentSlot.CHEST, new ItemStack(chestplate));
-        }
-        if (random.nextInt(100) < 50) {
-            setStackIfEmpty(entity, EntityEquipmentSlot.LEGS, new ItemStack(leggings));
-        }
-        if (random.nextInt(100) < 50) {
-            setStackIfEmpty(entity, EntityEquipmentSlot.FEET, new ItemStack(boots));
-        }
+    private void setEntityArmorWithChance(Random random, EntityLivingBase entity, ToolsItem sword, ToolsItem helmet, ToolsItem chestplate, ToolsItem leggings, ToolsItem boots) {
```

Comment:
Personally not a fan of this; the separate `if` statements make it clearer that it's a conditional thing, and without, the itemstack gets created whether it's used or not